Extras/Scripts/pre-commit-checks.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+LINT_CHECK_PATHS=(
+    "Common/"
+    "Signalling/"
+    "Frontend/library/"
+    "Frontend/ui-library/"
+    "SignallingWebServer/"
+)
+
+FAILURE=0
+FAILED_PATHS=()
+
+for CHECK_PATH in "${LINT_CHECK_PATHS[@]}"; do
+    STAGED_FILES=$(git diff --name-only --cached "$CHECK_PATH")
+
+    # check if any of the npm packages have changes
+    if [[ -n "$STAGED_FILES" ]]; then
+        echo ${CHECK_PATH} has changes. Running lint...
+        pushd $CHECK_PATH >/dev/null
+        if ! npm run lint; then
+            FAILURE=1
+            FAILED_PATHS+=("$CHECK_PATH")
+        fi
+        popd >/dev/null
+    fi
+done
+
+if [[ $FAILURE -eq 1 ]]; then
+    echo Linting failures in the following paths...
+    for FAILED_PATH in "${FAILED_PATHS[@]}"; do
+        echo "    $FAILED_PATH"
+    done
+    exit 1
+fi
+
+
```

Comment:
Maybe this could be in JS?

```js
const { execSync } = require('child_process');

const LINT_CHECK_PATHS = [
    'Common/',
    'Signalling/',
    'Frontend/library/',
    'Frontend/ui-library/',
    'SignallingWebServer/'
];

let failure = false;
const failedPaths = [];

LINT_CHECK_PATHS.forEach((checkPath) => {
    try {
        // Check for staged files in the given path
        const stagedFiles = execSync(`git diff --name-only --cached ${checkPath}`).toString().trim();

        if (stagedFiles) {
            console.log(`${checkPath} has changes. Running lint...`);

            // Run lint command
            try {
                execSync('npm run lint', { cwd: checkPath, stdio: 'inherit' });
            } catch {
                failure = true;
                failedPaths.push(checkPath);
            }
        }
    } catch (err) {
        console.error(`Error checking path: ${checkPath}`, err);
    }
});

if (failure) {
    console.error('Linting failures in the following paths:');
    failedPaths.forEach((path) => console.error(`    ${path}`));
    process.exit(1);
}
```

Reply:
Nvm seems Windows git always comes with bash:

![image](https://github.com/user-attachments/assets/2b5c5867-f70b-4599-8ae8-db16039da3f2)